sway-core/src/semantic_analysis/ast_node/expression/typed_expression.rs

```diff
@@ -1188,6 +1184,16 @@ impl ty::TyExpression {
             (false, None, None, Some((const_decl, span))) => {
                 warnings.append(&mut const_probe_warnings);
                 errors.append(&mut const_probe_errors);
+                if !unknown_call_path_binding.type_arguments.to_vec().is_empty() {
```

Comment:
I don't think you should use `unknown_call_path_binding` here given we know it's for a const declaration at that point. I remember setting the name specifically to distinguish it from the later moves. Better to use the same pattern as the function and pass it through the pattern as a tuple that you get to call `call_path_binding`